ls1intum/Pyris
# Add checkhealth endpoint

## Description
- Protect function call with circuit breaker: 
  + if the circuit is open, a function call will raise "too many failures" error
  + if the circuit is closed, perform the function call will:
    ++ if call successfully, set the circuit status to closed and cache this result for 5 minutes
    ++ if call unsuccessfully, set the circuit status to open and cache this result for 30 seconds
- Add health endpoint to return UP (the circuit is closed) and DOWN (the circuit is open) status for each model. This endpoint is called by Artermis regularly, so it will try to get the status from the cache before calling to get new status

## Step for testing
- Purposely set wrong config for one model or just use the values from `application.example.yml`
- Call `POST /api/v1/messages` for 3 times for that model, and the fourth time will have "too many failures" error
- Wait for 30 seconds and can continue to call that model
- Call `GET /api/v1/health` and should have the correct status for the models
